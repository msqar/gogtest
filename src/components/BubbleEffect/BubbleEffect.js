import React from 'react';

class BubbleEffect extends React.PureComponent {
    render() {
        return (
            <div className="BubbleEffect">
                <div className="BubbleEffect-circle"></div>
            </div>
        )
    }
}

export default BubbleEffect;