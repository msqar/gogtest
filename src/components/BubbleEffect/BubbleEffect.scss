@import '../../styles/_palette.scss';

.BubbleEffect {
    width: 80px;
    height: 80px;
    display: flex;
    align-items: center;
    justify-content: center;
    overflow: hidden;
    position: absolute;
    left: -4px;
    -webkit-transition: color 0.7s;
    transition: color 0.7s;
    -webkit-tap-highlight-color: rgba(0, 0, 0, 0);
}

.BubbleEffect-circle {
    &:before,
    &:after {
        box-shadow: 0 0 0 2px $green-200;
    }

    &:before {
        position: absolute;
        top: 50%;
        left: 50%;
        margin: -35px 0 0 -35px;
        width: 70px;
        height: 70px;
        border-radius: 50%;
        content: '';
        opacity: 0;
        pointer-events: none;
        -webkit-animation: anim-bubble-1 0.5s forwards;
        animation: anim-bubble-1 0.5s forwards;
    }

    &:after {
        -webkit-animation: anim-bubble-2 0.5s forwards;
        animation: anim-bubble-2 0.5s forwards;
    }
}

@keyframes anim-bubble-1 {
	0% {
		opacity: 1;
		-webkit-transform: scale3d(0.5, 0.5, 1);
		transform: scale3d(0.5, 0.5, 1);
	}
	100% {
		opacity: 0;
		-webkit-transform: scale3d(1.1, 1.1, 1);
		transform: scale3d(1.1, 1.1, 1);
	}
}

@keyframes anim-bubble-2 {
	0% {
		opacity: 1;
		-webkit-transform: scale3d(0.5, 0.5, 1);
		transform: scale3d(0.5, 0.5, 1);
	}
	50%, 100% {
		opacity: 0;
		-webkit-transform: scale3d(1.2, 1.2, 1);
		transform: scale3d(1.2, 1.2, 1);
	}
}