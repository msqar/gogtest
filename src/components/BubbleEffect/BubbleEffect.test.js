import React from 'react';
import renderer from 'react-test-renderer';
import BubbleEffect from './BubbleEffect';

it('renders BubbleEffect correctly', () => {
    const tree = renderer.create(<BubbleEffect />).toJSON();
    expect(tree).toMatchSnapshot();
});