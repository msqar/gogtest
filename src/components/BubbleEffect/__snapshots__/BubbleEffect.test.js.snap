// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders BubbleEffect correctly 1`] = `
<div
  className="BubbleEffect"
>
  <div
    className="BubbleEffect-circle"
  />
</div>
`;
